nupic/regions/KNNClassifierRegion.py

```diff
@@ -551,12 +555,157 @@ def __init__(self,
     self._tapFileOut = None
 
     self._initEphemerals()
+    self._knn = KNNClassifier.KNNClassifier(**self.knnParams)
```

Comment:
Looks like it was constructed in _initEphemerals. How about we leave it there but declare it here (before call to _initEphemerals):

```
# Declare _knn that is set in _initEphemerals
self._knn = None

self._initEphemerals()
```